tmt/steps/__init__.py

```diff
@@ -242,21 +363,55 @@ def wake(self) -> None:
         # order to cover a very frequent use case 'tmt run --last report'
         # FIXME find a better way how to enable always-force per plugin
         if (isinstance(self, tmt.steps.report.Report) and
-                self.data[0].get('how') in ['display', 'html']):
+                self.data[0].how in ['display', 'html']):
             self.debug("Report step always force mode enabled.")
             self._workdir_cleanup()
             self.status('todo')
 
         # Nothing more to do when the step is already done
         if self.status() == 'done':
+            self.debug('step is done, not touching its data')
             return
 
         # Override step data with command line options
         how: str = self.opt('how')
         if how is not None:
-            for data in self.data:
-                assert isinstance(data, dict)
-                data['how'] = how
+            # If 'how' has been given, when it comes to current entries in `self.data`,
+            # there are two options:
+            #
+            # * entry's `how` is the same as the one given via command-line. Then we can
+            # keep step data we already have.
+            # * entry's `how` is different, and then we need to throw the entry away and
+            # replace it with new `how`.
+            #
+            # To handle both variants, we replace `self.data` with new set of entries,
+            # based on newly constructed set of raw data.
+            self.debug(f'CLI-provided how={how} overrides all existing step data', level=4)
+
+            _raw_data: List[_RawStepData] = []
+
+            for datum in self.data:
+                # We can re-use this one - to make handling easier, just dump it to "raw"
+                # form for _normalize_data().
+                if datum.how == how:
+                    self.debug(f'  compatible:   {datum}', level=4)
+                    _raw_data.append(datum.to_raw())
+
+                # Mismatch, throwing away, replacing with new `how` - btu we can keep the name.
```

Comment:
```suggestion
                # Mismatch, throwing away, replacing with new `how` - but we can keep the name.
```

Reply:
Ack, will fix.

Reply:
Fixed in [8d096fd](https://github.com/teemtee/tmt/pull/1461/commits/8d096fd6dedfb1f4dab9cea49f75946075af5836)